### PR TITLE
firestore: Implement `put*` methods

### LINKED ISSUE #47: Implement put{Picture,Category} for Firestore back-end
follow-up to #21

### PR DESCRIPTION
**Pull request by bbjubjub2494**
Closes #47 

- [x] create and configure the scratch database in Firebase for unit tests
- [x] write unit tests

### CODE REVIEW

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatasetInterface.kt [hunk 1]
```diff
@@ -1,38 +1,56 @@
...
 import com.google.firebase.ktx.app
+import com.google.firebase.storage.ktx.storage
 import kotlinx.coroutines.tasks.await
-import java.io.Serializable
+import java.util.*
 
 class FirestoreDatasetInterface(
+    /**
+     * name of a database within the Firebase App
+     */
```
Comment (wattsecs) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatasetInterface.kt:19]: What do you mean by "within the Firebase App"? Is it not the same as the name give to the dataset (for example "Cooking ustensils")

Comment (bbjubjub2494) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatasetInterface.kt:19]: They are indeed different. My situation is that there can be separate databases in the same Firebase project, like test and production. (Currently we have 2 different test databases, see firebase-setup.md) The "DatasetInterface" needs to select one. The name of a dataset is associated with a dataset, which is an entity within the database. However, we currently operate under the assumption that there is only one dataset in a database. (see #50)

Comment (wattsecs) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatasetInterface.kt:19]: Then I would suggest renaming the class to `FirestoreDatabaseInterface`

**Review by wattsecs — APPROVED** [review]
Approved following offline discussion.

**Review by martin-lenweiter — APPROVED** [review]
Approved for same reason as Jonas

**Pull request merged**